Secret-agency-persistence/src/main/java/cz/fi/muni/pa165/secretagency/dao/AgentDaoImpl.java

```diff
@@ -0,0 +1,44 @@
+package cz.fi.muni.pa165.secretagency.dao;
+
+import cz.fi.muni.pa165.secretagency.entity.Agent;
+import cz.fi.muni.pa165.secretagency.enums.AgentRankEnum;
+import cz.fi.muni.pa165.secretagency.enums.LanguageEnum;
+
+import java.time.LocalDate;
+import java.util.List;
+
+public class AgentDaoImpl extends GenericDaoImpl<Agent> implements AgentDao {
+
+    /**
+     * Constructor
+     * Sets this DaoImpl to work with CRUD operations from generic implementation.
+     */
+    public AgentDaoImpl() {super(Agent.class);}
+
+    @Override
+    public List<Agent> getAgentsWithLanguageKnowledge(LanguageEnum language) {
+        return em.createQuery("SELECT a FROM Agent a WHERE :lang MEMBER OF a.languages", Agent.class)
+                .setParameter("lang", language).getResultList();
```

Comment:
Pojmenovával bych parametr v query stejně, jako je parametr metody. Zdá se mi to přehlednější. Ale jde jen o preference.
V setParemeter bys pak měl ("language", language), v dané query taky a v hlavičce metody též. Snadněji se pak odhalí překlep,...